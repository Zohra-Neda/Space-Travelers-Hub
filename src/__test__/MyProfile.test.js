import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MyProfile from '../pages/MyProfile';

const mockStore = configureStore([]);

describe('MyProfile component', () => {
  test('renders correctly', () => {

    const initialState = {
      rockets: {
        value: [
          {
            id: 'rocket1',
            name: 'Rocket 1',
            reserved: true,
          },

        ],
      },
      missions: {
        value: [
          {
            id: 'mission1',
            name: 'Mission 1',
            description: 'This is the mission description',
            joined: true,
          },

        ],
      },
    };
    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
