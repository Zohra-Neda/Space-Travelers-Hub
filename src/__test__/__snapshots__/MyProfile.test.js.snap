// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile component renders correctly 1`] = `
<div
  className="flex justify-center my-10"
>
  <div
    className="profile-container flex gap-5 w-[80%]"
  >
    <div
      className="w-[50%]"
    >
      <h2
        className="text-4xl font-semibold"
      >
        My Missions
      </h2>
      <div
        className="my-mission-container my-5"
      >
        <div
          className="my-mission-card border border-1  text-2xl ps-4 py-3 "
        >
          Mission 1
        </div>
      </div>
    </div>
    <div
      className="w-[50%]"
    >
      <h2
        className="text-4xl font-semibold"
      >
        My Rockets
      </h2>
      <div
        className="my-rockets-container my-5"
      >
        <div
          className="my-rocket-card border border-1 py-3 text-2xl ps-4 "
        >
          Rocket 1
        </div>
      </div>
    </div>
  </div>
</div>
`;
